### PR TITLE
API/UCP: Rename listener functions and parameters.

### PR DESCRIPTION
**Pull request by alinask**

### CODE REVIEW

**Comment by yosefe** [comment]
@shamisp @bbenton pls take a look

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5715/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5726/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5727/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shamisp @bbenton pls take a look

**Comment by shamisp** [comment]
1. why we are doing this ?
2. Not backward compatible.

**Comment by yosefe** [comment]
1. To be compatible with naming of other functions
2. This API was not part of v1.2, so better fix it now before v1.3 is released

**Comment by shamisp** [comment]
👍

File: src/ucp/api/ucp.h [hunk 1]
```diff
@@ -770,20 +771,20 @@ typedef struct ucp_worker_listener_params {
...
+     * The @ref ucp_listener_create routine will return with an error if sockaddr
      * is not specified.
      */
     ucs_sock_addr_t                sockaddr;
 
     /**
      * Handler to endpoint creation in a client-server connection flow.
      * In order for the callback inside this handler to be invoked, the
-     * UCP_WORKER_LISTENER_PARAM_FIELD_CALLBACK needs to be set in the
+     * UCP_LISTENER_PARAM_FIELD_CALLBACK needs to be set in the
```
Comment (bbenton) [inline src/ucp/api/ucp.h]: This should be UCP_LISTENER_PARAM_FIELD_ACCEPT_HANDLER. Also, it should have  `@ref``.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5749/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5754/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5757/ for details (Mellanox internal link).

**Comment by bbenton** [comment]
:+1: for Doc/API

**Pull request merged**